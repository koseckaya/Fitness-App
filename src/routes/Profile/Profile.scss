@import '../../styles/vars.scss';
@import '../../styles/light-dark-theme';

.tab-menu__wrapper {
  margin: 0 auto;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.tab-menu > button:not(:last-child) {
  margin-right: 1rem;
}

.tab-menu > button {
  cursor: pointer;
  padding: 8px 16px;
  border: 0;
  border-bottom: 2px solid transparent;
  background: none;
  font-size: 16px;
  color: var(--textThemeColor1);
  font-family: "Manrope", sans-serif;
}

.button-active {
  border-bottom: 2px solid #007bef;
  font-weight: 600;
}

.tab-menu > button:hover {
  border-bottom: 2px solid #007bef;
}

.tab-content {
  display: none;
}

.tab-content.selected {
  display: block;
}

.tab-view {
  width: 100%;
}