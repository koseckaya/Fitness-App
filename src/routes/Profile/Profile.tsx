import '../../components/Signup-Form/signup-form.scss';
import '../../components/Button/Button.scss'

import { FC, useContext, useEffect, useState} from 'react';
import { useForm } from 'react-hook-form';
import { yupResolver } from '@hookform/resolvers/yup';
import * as Yup from 'yup';
import { 
  AdditionalInformation,
  getUserDocFromAuth
} from '../../components/utils/firebase/firebase';
import { User } from 'firebase/auth';
import { UserContext } from '../../components/utils/contexts';

type UserSubmitForm = {
  name: string;
  lastName?: string;
};

export const Profile: FC = () => {
  const { currentUser } = useContext(UserContext);
  const userData = useUserData(currentUser);

  const validationSchema = Yup.object().shape({
    name: Yup.string().optional()
      .min(3, 'Name must be at least 4 characters')
      .max(20, 'Name must not exceed 20 characters'),
    lastName: Yup.string().optional()
      .min(3, 'Name must be at least 4 characters')
      .max(20, 'Name must not exceed 20 characters'),
  });

  const {
    register,
    handleSubmit,
    reset,
    formState: { errors },
  } = useForm<UserSubmitForm>({
    resolver: yupResolver(validationSchema)
  });

  const onSubmit = async (data: UserSubmitForm) => {
    if (!data) return;
/*  try {
      const response = await createAuthUserWithEmailAndPass(data.email, data.password);
      const user = response?.user;
      if (user) await createUserDocFromAuth(user, { displayName: data.name, lastName: data.lastName });
    } catch (error) {
      if (error instanceof Error && error.code === 'auth/email-already-in-use') {
        console.log('User already in use');
      }
      console.log('user create encountered an error', error);
    } */
    reset();
  };

  return (
    <div className='form-wrap'>
      <div className='form-container'>
        <h2 className='form-title'>Personal Info</h2>
        <form className='form__auth'
          onSubmit={handleSubmit(onSubmit)}>
          <p>Your Name</p>
          <div>
            <div className="mb-3">
              <input type="text" id="name" aria-describedby="nameHelp"
                className={`form-control__auth ${ errors.name ? 'is-invalid' : '' }`}
                placeholder='First Name'
                defaultValue={userData?.displayName}
                {...register('name')} />
              <p className='invalid-feedback'>{errors.name?.message}</p>
            </div>
            
            <div className="mb-3">
              <input type="text" id="lastName" aria-describedby="nameHelp"
                className={`form-control__auth ${ errors.lastName ? 'is-invalid' : '' }`}
                placeholder='Last Name'
                defaultValue={userData?.lastName}
                {...register('lastName')} />
              <p className='invalid-feedback'>{errors.lastName?.message}</p>
            </div>
          </div>
          <div>
            <button type='button' className='button form-btn'>
              Cancel
            </button>
            <button type='submit' className='button form-btn'>
              Save Changes
            </button>
          </div>
        </form>
      </div>
    </div>
  );
}

export function useUserData(currentUser: User | null) {
  const [data, setUserData] = useState<AdditionalInformation | null>(null);
  
  useEffect(() => {
    async function startFetching() {
      setUserData(null);
      try {
        if (!currentUser) return;
        const result = await getUserDocFromAuth(currentUser);
        if (!ignore && result) {
          setUserData(result);
        } 
      } catch (error) {
        if (error instanceof Error) {
          console.log('Error with get user', error.code);
        }
      }
    }

    let ignore = false;
    startFetching();
    return () => {
      ignore = true;
    }
  }, [currentUser]);

  return data;
}

export default Profile;