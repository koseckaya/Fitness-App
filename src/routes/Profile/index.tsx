import {Profile, useUserData} from "./Profile";

export {Profile, useUserData};