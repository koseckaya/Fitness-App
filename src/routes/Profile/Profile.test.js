import React from 'react';
import renderer from 'react-test-renderer';
import { MemoryRouter as Router } from 'react-router-dom';
import Profile from './Profile';


it('renders correctly', () => {
    const tree = renderer
        .create(<Router><Profile /></Router>)
        .toJSON();
    expect(tree).toMatchSnapshot();
});