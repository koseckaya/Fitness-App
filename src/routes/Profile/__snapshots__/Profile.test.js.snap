// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container "
>
  <div
    className="tab-menu__wrapper"
  >
    <div
      className="tab-menu"
    >
      <button
        className="focus"
        onClick={[Function]}
      >
        Personal Info
      </button>
      <button
        className=""
        onClick={[Function]}
      >
        Started Challenges
      </button>
    </div>
    <div
      className="tab-view"
    >
      <div
        className="tab-content selected"
        index="1"
        label="Personal Info"
      >
        <div
          className="form-wrap"
        >
          <div
            className="form-container"
          >
            <form
              className="form__auth"
              onSubmit={[Function]}
            >
              <p
                className="form__auth-subtitle"
              >
                Your Name
              </p>
              <div>
                <div
                  className="mb-3"
                >
                  <input
                    aria-describedby="nameHelp"
                    className="form-control__auth "
                    id="name"
                    name="name"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="First Name"
                    type="text"
                  />
                  <p
                    className="invalid-feedback"
                  />
                </div>
                <div
                  className="mb-3"
                >
                  <input
                    aria-describedby="nameHelp"
                    className="form-control__auth "
                    id="lastName"
                    name="lastName"
                    onBlur={[Function]}
                    onChange={[Function]}
                    placeholder="Last Name"
                    type="text"
                  />
                  <p
                    className="invalid-feedback"
                  />
                </div>
              </div>
              <div
                className="form-btn__container"
              >
                <button
                  className="button form-btn form-btn__default"
                  disabled={true}
                  onClick={[Function]}
                  type="button"
                >
                  Cancel
                </button>
                <button
                  className="button form-btn"
                  disabled={true}
                  type="submit"
                >
                  Save Changes
                </button>
              </div>
              <div>
                <p
                  className="form__auth-subtitle"
                >
                  Email Address
                </p>
                <p
                  className="auth__text"
                >
                  Your email address is
                  <span
                    className="auth__text-secondary"
                  />
                </p>
              </div>
              <div>
                <p
                  className="form__auth-subtitle"
                >
                  Account created
                </p>
                <p
                  className="auth__text"
                >
                  Time of creation
                  <span
                    className="auth__text-secondary"
                  />
                </p>
              </div>
            </form>
          </div>
        </div>
      </div>
      <div
        className="tab-content "
        index="2"
        label="Started Challenges"
      >
        <div
          className="small-programs__container"
        >
          <p>
            Start Workout Programs
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
